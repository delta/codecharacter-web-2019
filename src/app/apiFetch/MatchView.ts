/* tslint:disable:no-console*/
import { API_BASE_URL } from '../../config/config';

export const getMatches = () => {
  return fetch(`${API_BASE_URL}match/all`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};

export const getTopMatches = () => {
  return fetch(`${API_BASE_URL}match/pro`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};

export const getGameLogs = (gameId: number) => {
  return fetch(`${API_BASE_URL}match/log/${gameId}`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => response.json())
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};
