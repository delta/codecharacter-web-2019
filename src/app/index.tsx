import Dashboard from 'app/containers/Dashboard';
// @ts-ignore
import { initializeRendererAssets } from 'codecharacter-renderer';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

initializeRendererAssets();

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
