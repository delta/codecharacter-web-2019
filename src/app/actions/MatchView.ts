import { Match } from 'app/types/MatchView';
import { action } from 'typesafe-actions';

export namespace MatchActions {
  export enum Type {
    GET_MATCHES = 'GET_MATCHES',
    GET_TOP_MATCHES = 'GET_TOP_MATCHES',
    VIEW_MATCH = 'VIEW_MATCH',
    UPDATE_LOADING_STATUS = 'UPDATE_LOADING_STATUS',
    UPDATE_ERROR = 'UPDATE_ERROR',
    UPDATE_MATCHES = 'UPDATE_MATCHES',
    UPDATE_TOP_MATCHES = 'UPDATE_TOP_MATCHES',
  }

  export const getMatches = () => action(Type.GET_MATCHES);

  export const getTopMatches = () => action(Type.GET_TOP_MATCHES);

  export const updateLoadingStatus = (loadingStatus: boolean) => {
    return action(Type.UPDATE_LOADING_STATUS, { loadingStatus });
  };

  export const updateMatches = (matchesData: Match[]) => {
    return action(Type.UPDATE_MATCHES, { matchesData });
  };

  export const updateTopMatches = (topMatchesData: Match[]) => {
    return action(Type.UPDATE_TOP_MATCHES, { topMatchesData });
  };

  export const updateError = (err: string) => {
    return action(Type.UPDATE_ERROR, err);
  };
}
