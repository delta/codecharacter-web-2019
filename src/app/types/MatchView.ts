import { MatchActions } from 'app/actions';
import { ActionType } from 'typesafe-actions';

export enum MatchViewTabType {
  MY_MATCHES = 'MY_MATCHES',
  TOP_MATCHES = 'TOP_MATCHES',
}

export interface OwnState {
  activeMatchViewTab: MatchViewTabType;
  offset: number;
}

export interface Match {
  userOne: string;
  userOneScore: number;
  userTwoScore: number;
  userTwo: string;
  winner: string;
  playedAt: Date;
}

export interface StateProps {
  matches: Match[];
  topMatches: Match[];
  loading: boolean;
}

export interface DispatchProps {
  getMatches: () => void;
  getTopMatches: () => void;
}

export type Props = StateProps & DispatchProps;

export type State = OwnState;

export interface ElementOwnProps {
  match: Match;
  index: number;
}

export type ElementProps = ElementOwnProps;

const actions = {
  getMatches: MatchActions.getMatches,
  getTopMatches: MatchActions.getTopMatches,
  updateError: MatchActions.updateError,
  updateLoading: MatchActions.updateLoadingStatus,
  updateMatches: MatchActions.updateMatches,
  updateTopMatchs: MatchActions.updateTopMatches,
};

export interface MatchStoreState {
  loading: boolean;
  matches: Match[];
  topMatches: Match[];
}

export type MatchStoreAction = ActionType<typeof actions>;
