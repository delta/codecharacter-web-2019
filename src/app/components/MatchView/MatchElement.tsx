import { faShieldAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as styles from 'app/styles/MatchView.module.css';
import { Avatar } from 'app/types/Authentication/Register';
import * as MatchInterfaces from 'app/types/MatchView';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';

export class MatchElement extends React.Component<
  MatchInterfaces.ElementProps,
  MatchInterfaces.ElementState
> {
  public MAP_THUMBNAIL = [
    'assets/img/maps/islesofcodechar.png',
    'assets/img/maps/timeturner.png',
    'assets/img/maps/treacherousmangroves.png',
    'assets/img/maps/mysterymap.png',
    'assets/img/maps/mysterymap.png',
  ];

  public constructor(props: MatchInterfaces.ElementProps) {
    super(props);
    this.state = {
      isHoveredOver: false,
    };
  }

  public render() {
    const { match, getGameLogs, currentUserMatch, type } = this.props;
    const { isHoveredOver } = this.state;

    const isMyMatch = type === MatchInterfaces.MatchViewTabType.MY_MATCHES;
    let matchResult = 'TIE';

    if (match.verdict === '0') {
      matchResult = 'TIE';
    } else if (match.verdict === '1') {
      if (currentUserMatch) matchResult = 'WON';
      else matchResult = 'LOST';
    } else {
      if (currentUserMatch) matchResult = 'LOST';
      else matchResult = 'WON';
    }

    return (
      <Col
        sm={12}
        style={{ borderRadius: 5 }}
        className={classnames('mb-1')}
        onMouseEnter={() => {
          this.setState({
            isHoveredOver: true,
          });
        }}
        onMouseLeave={() => {
          this.setState({
            isHoveredOver: false,
          });
        }}
      >
        <div
          className={classnames(
            'w-100',
            styles.matchElement,
            type === MatchInterfaces.MatchViewTabType.TOP_MATCHES ? styles.topMatchElement : '',
            {
              [`${styles.winBackground}`]: isMyMatch && matchResult === 'WON',
              [`${styles.loseBackground}`]: isMyMatch && matchResult === 'LOST',
            },
          )}
        >
          <Row className="d-flex justify-content-around ">
            <Col sm={5} style={{ textAlign: 'center' }}>
              <p style={{ fontSize: 13 }}>{`${match.username1.substr(0, 15)}${
                match.username1.length > 15 ? '...' : ''
              }`}</p>
            </Col>
            <Col sm={2} style={{ textAlign: 'center' }}>
              <FontAwesomeIcon icon={faShieldAlt} />
            </Col>
            <Col sm={5} style={{ textAlign: 'center' }}>
              <p style={{ fontSize: 13 }}>{`${match.username2.substr(0, 15)}${
                match.username2.length > 15 ? '...' : ''
              }`}</p>
            </Col>
          </Row>
          <div className={classnames(' d-flex justify-content-center w-100')}>
            {
              // @ts-ignore
              <img width={35} height={35} src={Avatar[match.avatar1]} />
            }
            <span className="text-capitalize text-font-weight-bold  h3 mx-3">
              {match.score1} - {match.score2}
            </span>
            {
              // @ts-ignore
              <img width={35} height={35} src={Avatar[match.avatar2]} />
            }
          </div>
          {isMyMatch ? (
            <div className=" d-flex justify-content-center text-capitalize text-font-weight-bold ">
              {matchResult}
            </div>
          ) : null}
          <div className={classnames(styles.body, 'd-flex justify-content-around ')}>
            {match.games.map((game, index) => (
              <span
                key={index}
                className={classnames(styles.game)}
                onClick={() => getGameLogs(game.id)}
              >
                <img
                  src={this.MAP_THUMBNAIL[game.mapId - 1]}
                  width={20}
                  height={20}
                  style={{
                    boxShadow: `0 0 20px ${
                      game.verdict === '0'
                        ? 'yellow'
                        : game.verdict === '1'
                          ? 'green'
                          : game.verdict === '2'
                            ? 'red'
                            : 'gray'
                      }`,
                  }}
                />
              </span>
            ))}
          </div>
          {isHoveredOver ? (
            <div
              className=" d-flex justify-content-start text-capitalize  my-1 mx-3 h6"
              style={{
                fontSize: '12px',
              }}
            >
              {new Date(match.playedAt).toLocaleString()}
            </div>
          ) : null}
        </div>
      </Col>
    );
  }
}
