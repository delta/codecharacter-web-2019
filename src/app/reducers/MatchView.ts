import { MatchActions } from 'app/actions';
import * as MatchViewInterfaces from 'app/types/MatchView';

const matchesInitialState: MatchViewInterfaces.MatchStoreState = {
  loading: false,
  matches: [],
  topMatches: [],
};

export const matchesReducer = (
  state = matchesInitialState,
  action: MatchViewInterfaces.MatchStoreAction,
) => {
  switch (action.type) {
    case MatchActions.Type.UPDATE_LOADING_STATUS: {
      return {
        ...state,
        loading: action.payload.loadingStatus,
      };
    }
    case MatchActions.Type.UPDATE_MATCHES: {
      return {
        ...state,
        loading: false,
        matches: action.payload.matchesData,
      };
    }
    case MatchActions.Type.UPDATE_TOP_MATCHES: {
      return {
        ...state,
        loading: false,
        topMatches: action.payload.topMatchesData,
      };
    }
    case MatchActions.Type.UPDATE_ERROR:
      return {
        ...state,
        loading: false,
      };
    case MatchActions.Type.RESET_MATCH_VIEW:
      return {
        ...matchesInitialState,
      };
    default:
      return state;
  }
};
