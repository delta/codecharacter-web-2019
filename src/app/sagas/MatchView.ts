import { GameLogActions, MatchActions } from 'app/actions';
import * as MatchFetch from 'app/apiFetch/MatchView';
import { checkAuthentication } from 'app/sagas/utils';
import { resType } from 'app/types/sagas';
import { all, call, put, takeEvery } from 'redux-saga/effects';
import { ActionType } from 'typesafe-actions';

export function* getMatches(action: ActionType<typeof MatchActions.getMatches>) {
  try {
    yield put(MatchActions.updateLoadingStatus(true));
    const res = yield call(MatchFetch.getMatches);
    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(MatchActions.updateMatches(res.matchData));
    }
  } catch (err) {
    throw err;
  }
}

export function* getTopMatches(action: ActionType<typeof MatchActions.getTopMatches>) {
  try {
    yield put(MatchActions.updateLoadingStatus(true));
    const res = yield call(MatchFetch.getTopMatches);

    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(MatchActions.updateTopMatches(res.matchData));
    }
  } catch (err) {
    throw err;
  }
}

export function* getGameLogs(action: ActionType<typeof MatchActions.getGameLogs>) {
  try {
    const res = yield call(MatchFetch.getGameLogs, action.payload.gameId);

    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(GameLogActions.clearDisplayDebugLog());

      const logs = res.logs;
      const debugLog1 = logs.player1Log;
      const debugLog2 = logs.player2Log;
      const gameLog = logs.gameLog;
      const matchPlayerId = logs.matchPlayerId;

      yield put(GameLogActions.updateGameLog('', '', ''));
      yield put(GameLogActions.updateGameLog(debugLog1, debugLog2, gameLog));
      yield put(GameLogActions.updateMatchPlayerId(matchPlayerId));
    }
  } catch (err) {
    throw err;
  }
}

export function* matchSagas() {
  yield all([
    takeEvery(MatchActions.Type.GET_MATCHES, getMatches),
    takeEvery(MatchActions.Type.GET_TOP_MATCHES, getTopMatches),
    takeEvery(MatchActions.Type.GET_GAME_LOGS, getGameLogs),
  ]);
}
