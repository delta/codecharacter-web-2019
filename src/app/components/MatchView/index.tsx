import { faCaretLeft, faCaretRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { MatchElement } from 'app/components/MatchView/MatchElement';
import * as styles from 'app/styles/MatchView.module.css';
import * as MatchInterfaces from 'app/types/MatchView';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Grid, Row } from 'react-bootstrap';
// tslint:disable-next-line
import ReactPaginate from 'react-paginate';

export class Match extends React.Component<MatchInterfaces.Props, MatchInterfaces.State> {
  private static paginationSize = 15;
  constructor(props: MatchInterfaces.Props) {
    super(props);
    this.state = {
      activeMatchViewTab: MatchInterfaces.MatchViewTabType.MY_MATCHES,
      offset: 0,
    };
  }

  public componentDidMount() {
    this.props.getMatches();
    this.props.getTopMatches();
  }

  public handlePageClick = (data: { selected: number }) => {
    this.setState({
      offset: Math.ceil(data.selected * Match.paginationSize),
    });
  };

  public render() {
    const { activeMatchViewTab } = this.state;
    const { matches, topMatches } = this.props;

    return (
      <Grid fluid={true} className={classnames(styles.MatchView)}>
        <Row className="justify-content-between py-2 px-3">
          <Col className="text-light font-weight-bold my-auto">MATCHES</Col>
        </Row>
        <Row>
          <div className="col d-flex justify-content-center mb-3">
            <button
              className={classnames(styles.customBtn, {
                [`${styles.buttonActive}`]:
                  activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES,
              })}
              onClick={() => {
                this.toggleNotificationTab(MatchInterfaces.MatchViewTabType.MY_MATCHES);
                this.props.getMatches();
              }}
            >
              {' '}
              My Matches{' '}
            </button>
          </div>
          <div className="col d-flex justify-content-center mb-3">
            <button
              className={classnames(styles.customBtn, {
                [`${styles.buttonActive}`]:
                  activeMatchViewTab === MatchInterfaces.MatchViewTabType.TOP_MATCHES,
              })}
              onClick={() => {
                this.toggleNotificationTab(MatchInterfaces.MatchViewTabType.TOP_MATCHES);
                this.props.getTopMatches();
              }}
            >
              {' '}
              Top Matches{' '}
            </button>
          </div>
        </Row>
        <Row className={classnames('mb-2', styles.matchViewWrap)}>
          {activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES ? (
            matches && matches.length ? (
              matches.map((match, index) =>
                match ? (
                  index >= this.state.offset &&
                  index <= this.state.offset + Match.paginationSize - 1 ? (
                    <MatchElement match={match} index={index - 1} key={index - 1} />
                  ) : null
                ) : (
                  <div className="ml-5"> Nothing to show </div>
                ),
              )
            ) : (
              <div className="ml-5"> Nothing to show </div>
            )
          ) : topMatches && topMatches.length ? (
            topMatches.map((match, index) =>
              match ? (
                index >= this.state.offset &&
                index <= this.state.offset + Match.paginationSize - 1 ? (
                  <MatchElement match={match} index={index - 1} key={index - 1} />
                ) : null
              ) : (
                <div className="ml-5"> Nothing to show </div>
              ),
            )
          ) : (
            <div className="ml-5"> Nothing to show </div>
          )}
        </Row>
        <Row>
          <Col className="d-flex justify-content-center" style={{ width: '100vw', margin: '10px' }}>
            <ReactPaginate
              previousLabel={
                <span>
                  <FontAwesomeIcon icon={faCaretLeft} /> <FontAwesomeIcon icon={faCaretLeft} />
                </span>
              }
              nextLabel={
                <span>
                  <FontAwesomeIcon icon={faCaretRight} /> <FontAwesomeIcon icon={faCaretRight} />
                </span>
              }
              breakLabel={'...'}
              breakClassName={'break-me'}
              marginPagesDisplayed={1}
              pageClassName={'atag'}
              pageRangeDisplayed={2}
              pageCount={Math.ceil(matches.length / Match.paginationSize)}
              activeLinkClassName={'active'}
              onPageChange={this.handlePageClick}
              containerClassName={'pagination'}
              activeClassName={'active'}
            />
          </Col>
        </Row>
      </Grid>
    );
  }

  private toggleNotificationTab = (activeMatchViewTab: MatchInterfaces.MatchViewTabType) => {
    this.setState({
      activeMatchViewTab,
    });
  };
}
