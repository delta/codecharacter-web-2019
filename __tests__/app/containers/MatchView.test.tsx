import MatchView from 'app/containers/MatchView';
import { configureStore } from 'app/store';
import { mount } from 'enzyme';
import * as React from 'react';

describe('MatchView Container', () => {
  const { store } = configureStore();
  const wrapper = mount(<MatchView />, {
    context: {
      store,
    },
  });

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
