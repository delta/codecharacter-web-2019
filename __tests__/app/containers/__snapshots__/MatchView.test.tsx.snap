// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchView Container Should render 1`] = `
<Connect(Match)>
  <Match
    currentUsername=""
    getGameLogs={[Function]}
    getMatches={[Function]}
    getTopMatches={[Function]}
    loading={true}
    matches={Array []}
    topMatches={Array []}
  >
    <Grid
      bsClass="container"
      className=""
      componentClass="div"
      fluid={true}
    >
      <div
        className="container-fluid"
      >
        <Row
          bsClass="row"
          className="justify-content-between py-2 px-3"
          componentClass="div"
        >
          <div
            className="justify-content-between py-2 px-3 row"
          >
            <Col
              bsClass="col"
              className="text-light font-weight-bold my-auto"
              componentClass="div"
            >
              <div
                className="text-light font-weight-bold my-auto"
              >
                MATCHES
              </div>
            </Col>
          </div>
        </Row>
        <Row
          bsClass="row"
          componentClass="div"
        >
          <div
            className="row"
          >
            <div
              className="col d-flex justify-content-center mb-3"
            >
              <button
                className="undefined"
                onClick={[Function]}
              >
                 
                My Matches
                 
              </button>
            </div>
            <div
              className="col d-flex justify-content-center mb-3"
            >
              <button
                className=""
                onClick={[Function]}
              >
                 
                Top Matches
                 
              </button>
            </div>
          </div>
        </Row>
        <Row
          bsClass="row"
          className="mb-2"
          componentClass="div"
        >
          <div
            className="mb-2 row"
          >
            <div
              className="ml-5"
            >
               Nothing to show 
            </div>
          </div>
        </Row>
        <Row
          bsClass="row"
          componentClass="div"
        >
          <div
            className="row"
          >
            <Col
              bsClass="col"
              className="d-flex justify-content-center"
              componentClass="div"
              style={
                Object {
                  "margin": "10px",
                  "width": "100vw",
                }
              }
            >
              <div
                className="d-flex justify-content-center"
                style={
                  Object {
                    "margin": "10px",
                    "width": "100vw",
                  }
                }
              >
                <PaginationBoxView
                  activeClassName="active"
                  activeLinkClassName="active"
                  breakClassName="break-me"
                  breakLabel="..."
                  containerClassName="pagination"
                  disableInitialCallback={false}
                  disabledClassName="disabled"
                  marginPagesDisplayed={1}
                  nextClassName="next"
                  nextLabel={
                    <span>
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              192,
                              512,
                              Array [],
                              "f0da",
                              "M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z",
                            ],
                            "iconName": "caret-right",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      />
                       
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              192,
                              512,
                              Array [],
                              "f0da",
                              "M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z",
                            ],
                            "iconName": "caret-right",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      />
                    </span>
                  }
                  onPageChange={[Function]}
                  pageClassName="atag"
                  pageCount={0}
                  pageRangeDisplayed={2}
                  previousClassName="previous"
                  previousLabel={
                    <span>
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              192,
                              512,
                              Array [],
                              "f0d9",
                              "M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z",
                            ],
                            "iconName": "caret-left",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      />
                       
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              192,
                              512,
                              Array [],
                              "f0d9",
                              "M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z",
                            ],
                            "iconName": "caret-left",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      />
                    </span>
                  }
                >
                  <ul
                    className="pagination"
                  >
                    <li
                      className="previous disabled"
                    >
                      <a
                        aria-disabled="true"
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        role="button"
                        tabIndex="0"
                      >
                        <span>
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  192,
                                  512,
                                  Array [],
                                  "f0d9",
                                  "M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z",
                                ],
                                "iconName": "caret-left",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-caret-left fa-w-6 "
                              data-icon="caret-left"
                              data-prefix="fas"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 192 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                           
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  192,
                                  512,
                                  Array [],
                                  "f0d9",
                                  "M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z",
                                ],
                                "iconName": "caret-left",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-caret-left fa-w-6 "
                              data-icon="caret-left"
                              data-prefix="fas"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 192 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                        </span>
                      </a>
                    </li>
                    <li
                      className="next"
                    >
                      <a
                        aria-disabled="false"
                        onClick={[Function]}
                        onKeyPress={[Function]}
                        role="button"
                        tabIndex="0"
                      >
                        <span>
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  192,
                                  512,
                                  Array [],
                                  "f0da",
                                  "M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z",
                                ],
                                "iconName": "caret-right",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-caret-right fa-w-6 "
                              data-icon="caret-right"
                              data-prefix="fas"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 192 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                           
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  192,
                                  512,
                                  Array [],
                                  "f0da",
                                  "M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z",
                                ],
                                "iconName": "caret-right",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-caret-right fa-w-6 "
                              data-icon="caret-right"
                              data-prefix="fas"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 192 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                        </span>
                      </a>
                    </li>
                  </ul>
                </PaginationBoxView>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </Match>
</Connect(Match)>
`;
